src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,90 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  onSubmit: (value: CommentData) => void;
+  isSubmiting: boolean;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ onSubmit, isSubmiting }) => {
+  const [comment, setComment] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [hasNameError, setHasNameError] = useState(false);
+  const [hasEmailError, setHasEmailError] = useState(false);
+  const [hasBodyError, setHasBodyError] = useState(false);
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setComment(prevComment => ({
+      ...prevComment,
+      name: event.target.value,
+    }));
+    setHasNameError(false);
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setComment(prevComment => ({
+      ...prevComment,
+      email: event.target.value,
+    }));
+    setHasEmailError(false);
+  };
+
+  const handleBodyChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setComment(prevComment => ({
+      ...prevComment,
+      body: event.target.value,
+    }));
+    setHasBodyError(false);
+  };
+
+  const reset = () => {
+    setComment({
+      name: '',
+      email: '',
+      body: '',
+    });
+
+    setHasNameError(false);
+    setHasEmailError(false);
+    setHasBodyError(false);
```

Comment:
Instead of having three different states you can create just one single state `errors` and store an object with the same structure as your comment (name, email, body). It will be more readable and you will have cleaner code.
It's just a recommendation to your future work.